### PR TITLE
chore: attempt until searched nft

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2702

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/DecentralizedSearch/src/SNSAdaptor/PluginHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return (
+        <Stack flexDirection="row" justifyContent="space-between" alignItems="center" className={classes.wrapper}>
+            <Stack flexDirection="row" justifyContent="space-between" gap={0.5} alignItems="center">
+                <Icons.DecentralizedSearch size={24} />
+                <Typography color={theme.palette.maskColor.dark} fontWeight="bolder">
+                    <PluginI18NFieldRender field={base.name} pluginID={PluginID.DecentralizedSearch} />
+                </Typography>
+            </Stack>
+            <Box>
+                <Box className={classes.provider}>
```
Comment (guanbinrui) [inline packages/plugins/DecentralizedSearch/src/SNSAdaptor/PluginHeader.tsx]: Does the outside one reducible?

File: packages/mask/src/plugins/Trader/trending/useTrending.ts [hunk 2]
```diff
@@ -91,8 +98,24 @@ export function useTrendingById(
...
+    token?: Web3Helper.FungibleTokenScope<void, NetworkPluginID.PLUGIN_EVM>,
+) {
+    return {
+        ...trending?.coin,
+        decimals: trending?.coin.decimals || token?.decimals || 0,
+        contract_address:
+            searchedContractAddress ?? trending?.contracts?.[0]?.address ?? trending?.coin.contract_address,
+        chainId: expectedChainId ?? trending?.contracts?.[0]?.chainId ?? trending?.coin.chainId,
+    } as Coin
+}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trending/useTrending.ts]:
```suggestion
function createCoinFromTrending(
    trending?: TrendingAPI.Trending,
    expectedChainId?: ChainId,
    searchedContractAddress?: string,
    token?: Web3Helper.FungibleTokenScope<void, NetworkPluginID.PLUGIN_EVM>,
): Coin {
    return {
        ...trending?.coin,
        decimals: trending?.coin.decimals || token?.decimals || 0,
        contract_address:
            searchedContractAddress ?? trending?.contracts?.[0]?.address ?? trending?.coin.contract_address,
        chainId: expectedChainId ?? trending?.contracts?.[0]?.chainId ?? trending?.coin.chainId,
    }
}
```

File: packages/mask/src/plugins/Trader/trending/useTrending.ts [hunk 3]
```diff
@@ -91,8 +98,24 @@ export function useTrendingById(
...
+
 export function useCoinIdByAddress(address: string) {
-    return useAsyncRetry(() => {
+    return useAsyncRetry(async () => {
+        if (!address) return undefined
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trending/useTrending.ts]:
```suggestion
        if (!address) return
```

File: packages/plugins/DecentralizedSearch/src/SNSAdaptor/PluginHeader.tsx (hunk 1 above)
Comment (hanszhoou) [reply packages/plugins/DecentralizedSearch/src/SNSAdaptor/PluginHeader.tsx]: yes

Comment (hanszhoou) [reply packages/plugins/DecentralizedSearch/src/SNSAdaptor/PluginHeader.tsx]: done

File: packages/mask/src/plugins/Trader/trending/useTrending.ts [hunk 4]
```diff
@@ -63,26 +67,29 @@ export function useTrendingById(
     } = useAsync(async () => {
         if (!id) return null
         if (!currency) return null
+        if (chainList)
+            return attemptUntil(
+                chainList.map((chainId) => async () => {
+                    const trending = await PluginTraderRPC.getCoinTrending(chainId, id, currency, dataProvider).catch(
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trending/useTrending.ts]: Move to the API handling part.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**